### ISSUE TITLE
Baremetal instance stuck in starting state and timeout after 30 minutes.

### ISSUE THREAD

**Issue by yummydsky** [apache/cloudstack#10025]
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME
API

##### CLOUDSTACK VERSION
4.19.1.3

##### OS / ENVIRONMENT
Cloudstack 4.19.1.3 on Ubuntu 22.04LTS

##### SUMMARY
I use Ubuntu 22.04 LTS as ISO to deploy a bare metal instance, but it seems that bare metal instance stuck in starting state and timeout like issue https://issues.apache.org/jira/browse/CLOUDSTACK-9716
```
2024-12-02 06:24:44,311 DEBUG [c.c.b.n.BareMetalResourceBase] (DirectAgent-39:ctx-6061c3d0) (logid:47b584a0) still wait for baremetal provision done notification for vm[name:i-2-30-VM], current vm state is Starting
```
![截圖 2024-12-03 上午9 56 12](https://github.com/user-attachments/assets/54902e82-ef21-4edb-a0b9-5f48ef9eaa1c)

##### STEPS TO REPRODUCE
![截圖 2024-12-03 上午9 54 01](https://github.com/user-attachments/assets/5a7ee4ad-ddba-4b60-8e1c-12d3a4f38474)
![截圖 2024-12-03 上午9 54 25](https://github.com/user-attachments/assets/cf1d7ab4-ff04-4bdf-a2f0-1a711a1c94e3)
![截圖 2024-12-03 上午9 54 35](https://github.com/user-attachments/assets/14ea7fe4-6905-4865-8b12-1a4f35367632)

##### EXPECTED RESULTS
The bare metal instance can be deployed os successfully and booted up.

##### ACTUAL RESULTS
The bare metal stuck in starting state because there is no one to notify and change the vm state to `Running`

**Comment by DaanHoogland** [comment]
thanks @yummydsky , baremetal has not been well maintained for a long time. Have you got other bare-metal instances that are running fine?

**Comment by yummydsky** [comment]
> thanks @yummydsky , baremetal has not been well maintained for a long time. Have you got other bare-metal instances that are running fine?

No, there are no bare-metal instances that can be created properly.

**Comment by yummydsky** [comment]
@DaanHoogland Is there a PXE server inside the virtual router? I found a document(http://events17.linuxfoundation.org/sites/events/files/slides/Baremetal-v4.1a.pdf) that said we need an additional VR to act as a PXE server. Do you know how I can add a VR as a PXE server?

**Comment by DaanHoogland** [comment]
@yummydsky I do not know of this , but it seems that adding a pxe server itself to a VR is not a big deal. configuring it properly may vary depending of your infrastructure and requirements.
@rajujith Do you have any experience with this?
